toolchain/check/testdata/class/generic/redeclare.carbon

```diff
@@ -0,0 +1,208 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+
+// --- valid.carbon
+
+library "valid" api;
+
+class Generic(T:! type);
+
+class Generic(T:! type) {
+}
+
+// --- fail_invalid.carbon
```

Comment:
FWIW, you might consider splitting each of these to separate files, so that each is tested to independently fail. i.e., right now if one stops failing and we miss it in review, the others failing will lead to this test passing.

Reply:
Done.